src/query/service/src/interpreters/interpreter_table_set_options.rs

```diff
@@ -89,25 +89,20 @@ impl Interpreter for SetOptionsInterpreter {
         }
         let catalog = self.ctx.get_catalog(self.plan.catalog.as_str()).await?;
         let database = self.plan.database.as_str();
-        let table = self.plan.table.as_str();
-        let tbl = catalog
-            .get_table(self.ctx.get_tenant().as_str(), database, table)
+        let table_name = self.plan.table.as_str();
+        let table = match catalog
+            .get_table(self.ctx.get_tenant().as_str(), database, table_name)
             .await
-            .ok();
-
-        let table = if let Some(table) = &tbl {
-            // check mutability
-            table.check_mutable()?;
-            table
-        } else {
-            return Err(ErrorCode::UnknownTable(format!(
-                "Unknown table `{}`.`{}` in catalog '{}'",
-                database,
-                self.plan.table.as_str(),
-                &catalog.name()
-            )));
+        {
+            Ok(table) => table,
+            Err(error) => {
+                return Err(error);
+            }
```

Comment:
or just `let table = match catalog
            .get_table(self.ctx.get_tenant().as_str(), database, table_name)
            .await?;`

Reply:
well, this is my fault. Maybe after modify here many times to ensure the manner, so it causes this.